nextcord/http.py

```diff
@@ -151,48 +471,172 @@ def __exit__(
+
+        If ``True``, rate limits will use a discovered static-ish time delay to figure out when buckets reset.
+        Generally results in longer-than-necessary inaccurate bucket reset times, especially with high-latency
+        connections, but is most consistent across a wide variety of hosts and applications.
+
+        If ``False``, rate limits will use a combination of server-given bucket reset timestamps compared with local
+        time to accurately reset when the bucket resets and discovered static-ish time delays to predict the next reset
+        when needed.
+        Generally results in more accurate bucket reset times, especially with high-latency connections, but relies
+        heavily on host time being accurate with Discords time.
+    proxy: Optional[:class:`str`]
+        The proxy url to connect to the Discord API with, if any.
+    proxy_auth: Optional[:class:`aiohttp.BasicAuth`]
+        The authentication to use in order to make a request to the proxy url. Not to be confused with ``default_auth``.
+    dispatch: :class:`Callable`
+        The dispatcher to use for rate limit events.
+    """
 
     def __init__(
         self,
         connector: Optional[aiohttp.BaseConnector] = None,
         *,
+        max_global_requests: int = 50,
+        time_offset: float = 0.0,
+        default_auth: Optional[str] = None,
+        assume_unsync_clock: bool = False,
         proxy: Optional[str] = None,
         proxy_auth: Optional[aiohttp.BasicAuth] = None,
-        loop: Optional[asyncio.AbstractEventLoop] = None,
-        unsync_clock: bool = True,
         dispatch: Callable,
     ) -> None:
-        self.loop: asyncio.AbstractEventLoop = asyncio.get_event_loop() if loop is None else loop
-        self.connector = connector
         self.__session: aiohttp.ClientSession = MISSING  # filled in static_login
-        self._locks: weakref.WeakValueDictionary = weakref.WeakValueDictionary()
-        self._global_over: asyncio.Event = asyncio.Event()
-        self._global_over.set()
-        self.token: Optional[str] = None
-        self.bot_token: bool = False
-        self.proxy: Optional[str] = proxy
-        self.proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
-        self.use_clock: bool = not unsync_clock
-        self._dispatch: Callable = dispatch
+        self._connector = connector
+        self._max_global_requests = max_global_requests
+        self._time_offset = time_offset
+        self._default_auth = None
+        # For consistency with possible future changes to set_default_auth.
+        self.set_default_auth(default_auth)
+        self._ratelimit_use_timestamp = not assume_unsync_clock
+        self._proxy = proxy
+        self._proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
+        self._dispatch = dispatch
 
         # to mitigate breaking changes
-        self.user_agent: str = _USER_AGENT
+        self._user_agent: str = _USER_AGENT
```

Comment:
If it is to mitigate breaking changes maybe don't change the name?

Reply:
That comment was apart of #1114, unrelated to this PR.